packages/sdk/src/evm/contracts/smart-contract.ts

```diff
@@ -286,4 +292,11 @@ export class SmartContract<TContract extends BaseContract = BaseContract>
     }
     return undefined;
   }
+
+  private detectAppURI() {
+    if (detectContractFeature<AppURI>(this.contractWrapper, "AppURI")) {
```

Comment:
Here you should also check for the contract metadata extension

Reply:
so check for both? you can have appURI without contract metadata

Reply:
one OR the other.

`appUri = detect(AppUri) || detect(ContractMetadata)`

Reply:
@nachoiacovino @furqanrydhan ^